src/NuGet.Core/NuGet.Packaging/NupkgMetadata/NupkgMetadataFileFormat.cs

```diff
@@ -57,6 +57,11 @@ public static NupkgMetadataFile Read(TextReader reader, ILogger log, string path
                     Strings.Error_LoadingHashFile,
                     path, ex.Message));
 
+                if (!string.IsNullOrEmpty(path) && ex is InvalidDataException)
+                {
+                    throw new InvalidDataException(message: path, innerException: ex);
```

Comment:
Why only when the exception is InvalidDataException?  What if a different process had the file locked, preventing sharing with other processes? Or if some crazy process delete the file after NuGet has performed the `if File.Exists(path)` check, but before NuGet actually reads the file?  Isn't the filename still interesting to customers?

Reply:
Are you saying catch `Exception` and rethrow all of them as inner exceptions with a message? I don't really have time this sprint to test every scenario, like what will the output be from your example, and whether the logger already shows something or we should rethrow exceptions or not. Therefore, if that's necessary, I'll probably close this PR and let someone plan the work to encompass all scenarios.

Reply:
Yes, I'd just catch all exceptions. Regardless of what the root cause (file locked, invalid json content, file does not exist), I believe it's useful for customers to know what file had an issue, so it's not intuitive to me why we'd only report filename on specific exception types. Also, this code isn't trying to recover from an exception, it' just immediately throwing a new exception with additional information (retaining the original exception as an inner exception), so I don't think the general guideline of "catch specific exceptions" is very convincing here.

Also consider that `NupkgMetadataFileFormat.Read` is a public API that customers might use in their own apps. While we care mostly about our own tools experience, I think it would be a mistake to ignore the NuGet Client SDK experience.

Reply:
Is the invalid data exception always the one thrown at line 49?

Reply:
> Is the invalid data exception always the one thrown at line 49?

Yes. `throw new InvalidDataException();`